src/Taylor1.jl

```diff
@@ -666,3 +669,53 @@ function deriv{T<:Number}(a::Taylor1{T}, n::Int=1)
     @assert a.order >= n >= 0
     factorial( widen(n) ) * a.coeffs[n+1] :: T
 end
+
+
+"""
+    A_mul_B!(y, a, b)
+
+Multiply a*b and save the result in y.  Extends y to fit the results if necessary.
+"""
+function A_mul_B!{T<:Number}(y::Vector{Taylor1{T}},a::Matrix{T},b::Vector{Taylor1{T}})
+
+    # determine the maximal order of b
+    order = maximum([b1.order for b1 in b])
+
+    n,k = size(a)
+
+    # extend y to fit all the coefficients and zero the
```

Comment:
> Also I don't get what you mean by the naming convention and changing y. Can you explain it?

In Julia the convention is that functions ending with `!` imply that (at least) one of the arguments is changed. In this case it is `y`, which means that it must be already defined. In other words, it makes no sense to use this specialized formula if `y` is not already defined. If you test `A_mul_B!(z,A1,B)` using the definitions in the tests, it will through an error because `z` doesn't exist. So the `if` block in lines 700-705 is never executed.